### PR TITLE
Replace java.awt.geom.Rectangle2D.Double with ElkRectangle. 

### PR DESCRIPTION
**Pull request by jbeard4**
Fixes OpenKieler/elkjs#47

### CODE REVIEW

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/math/ElkMath.java [hunk 1]
```diff
@@ -1236,4 +1237,20 @@ public static double shortestDistance(final ElkRectangle r1, final ElkRectangle
...
+        double r1x1 = rect1.x;
+        double r1y1 = rect1.y;
+        double r1x2 = rect1.x + rect1.width;
+        double r1y2 = rect1.y + rect1.height;
+        double r2x1 = rect2.x;
+        double r2y1 = rect2.y;
+        double r2x2 = rect2.x + rect2.width;
+        double r2y2 = rect2.y + rect2.height;
+
+        return r1x1 < r2x2 && r1x2 > r2x1 && r1y1 > r2y2 && r1y2 < r2y1;
```
Comment (le-cds) [inline plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/math/ElkMath.java]: Maybe it's late, but shouldn't the last two parts be `r1y1 < r2y2 && r1y2 > r2y1`?

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/math/ElkMath.java [hunk 2]
```diff
@@ -1236,4 +1237,20 @@ public static double shortestDistance(final ElkRectangle r1, final ElkRectangle
         // case 3
         return -Math.sqrt(vertDist * vertDist + horzDist * horzDist);
     }
+
+    /**
+    * Tests if the interior of this rect intersects the interior of a specified rect
+    */
+    public static boolean intersectsRect(ElkRectangle rect1, ElkRectangle rect2){
```
Comment (le-cds) [inline plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/math/ElkMath.java]: Let's move this implementation directly into `ElkRectangle.intersects(...)`.

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/math/ElkMath.java (hunk 1 above)
Comment (jbeard4) [reply plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/math/ElkMath.java]: You are  correct. This is fixed in e46cf37

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/math/ElkMath.java (hunk 2 above)
Comment (jbeard4) [reply plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/math/ElkMath.java]: This is done. Thanks.

**Comment by le-cds** [comment]
Thanks a lot for this contribution and for addressing my comments, @jbeard4 :)

**Pull request merged**